src/aspire/source/__init__.py

```diff
@@ -370,6 +372,108 @@ def whiten(self, noise_filter):
         # Invalidate images
         self._im = None
 
+    def phase_flip(self):
+        """
+        Perform phase flip to images in the source object using CTF information
+        """
+        logger.info('Perform phase flip on source object')
+        logger.info('Adding Phase Flip Xform to end of generation pipeline')
+        self.generation_pipeline.add_xform(FlipXform(self.filters))
+        # Invalidate images
+        self._im = None
+
+    def reverse_contrast(self, batch_size=512):
```

Comment:
`invert_contrast` is a more standard name for this.

Reply:
Done.